src/network/networkd-link.c

```diff
@@ -1139,7 +1124,9 @@ static int link_set_proxy_arp(Link *link) {
         const char *p = NULL;
         int r;
 
-        if (!link_proxy_arp_enabled(link))
+        assert(link);
+
+        if (link->flags & IFF_LOOPBACK || !link->network || link->network->proxy_arp < 0)
```

Comment:
hmm, why remove the function that does this check? looks nicer with that function...

Also, if you mix binary and logical OR+AND in one expression, please use `()` to clarify when you use one and when the other... Or in other words, the first & check should really be in brackets...